conduit/pipeline.go

```diff
@@ -106,18 +108,11 @@ func MakePipelineConfig(logger *log.Logger, cfg *Config) (*PipelineConfig, error
 
 	logger.Infof("Auto-loading Conduit Configuration: %s", autoloadParamConfigPath)
 
-	file, err := os.Open(autoloadParamConfigPath)
-	if err != nil {
-		return nil, fmt.Errorf("MakePipelineConfig(): error opening file: %w", err)
-	}
-	defer file.Close()
-
-	err = viper.ReadConfig(file)
```

Comment:
viper doesn't support yaml tags that are different from the field name. for example, viper doesn't pick up `log-file` field when unmarshalling the config file.

Reply:
I think @AlgoStephenAkiki is more familiar with how we're using viper. I'm wondering if this breaks something that we were depending on--either with how you can set config values via environment variables or I recall discussing whether we accept `.yml` and `.yaml` extensions?

Reply:
Re: viper vs yaml - I think it just was a matter of preference from Will to consistently use viper instead of yaml.  We might need to get his opinion on that to go forward.  For the .yml/.yaml thing, I just had a list of extensions to check that I iterated through.

Reply:
I think I see part of the problem. The global viper config is used in `cmd/conduit/main.go`, I bet there are some weird side when we use it again here.

Earlier I was imagining we could use these features to simplify the pipeline init:
https://github.com/spf13/viper#extracting-a-sub-tree
https://github.com/spf13/viper#establishing-defaults

Since we're not using those, maybe we should remove viper completely and re-introduce it later if it ever makes sense.

Reply:
it would fail to load CPUProfile if the config file used `cpu-profile` instead of CPUprofile. it's the same problem for the other fields with kebab case tags, like PIDFilePath or PipelineLogLevel.